temporalcli/commands.workflow_exec.go

```diff
@@ -456,6 +475,91 @@ func (s *structuredHistoryIter) NextRawEvent() (*history.HistoryEvent, error) {
 	return event, nil
 }
 
+type eventFieldValue struct {
+	field string
+	value string
+}
+
+func (s *structuredHistoryIter) flattenFields(
+	cctx *CommandContext,
+	event *history.HistoryEvent,
+) ([]eventFieldValue, error) {
+	// We want all event fields and all attribute fields converted to the same
+	// top-level JSON object. First do the proto conversion.
+	opts := temporalproto.CustomJSONMarshalOptions{}
+	if cctx.JSONShorthandPayloads {
+		opts.Metadata = map[string]any{common.EnablePayloadShorthandMetadataKey: true}
+	}
+	protoJSON, err := opts.Marshal(event)
+	if err != nil {
+		return nil, fmt.Errorf("failed marshaling event: %w", err)
+	}
+	// Convert from string back to JSON
+	dec := json.NewDecoder(bytes.NewReader(protoJSON))
+	// We want json.Number
+	dec.UseNumber()
+	fieldsMap := map[string]any{}
+	if err := dec.Decode(&fieldsMap); err != nil {
+		return nil, fmt.Errorf("failed unmarshaling event proto: %w", err)
+	}
+	// Exclude eventId and eventType
+	delete(fieldsMap, "eventId")
+	delete(fieldsMap, "eventType")
+	// Lift any "Attributes"-suffixed fields up to the top level
+	for k, v := range fieldsMap {
+		if strings.HasSuffix(k, "Attributes") {
+			subMap, ok := v.(map[string]any)
+			if !ok {
+				return nil, fmt.Errorf("unexpectedly invalid attribute map")
+			}
+			for subK, subV := range subMap {
+				fieldsMap[subK] = subV
+			}
+			delete(fieldsMap, k)
+		}
+	}
+	// Flatten JSON map and sort
+	fields, err := s.flattenJSONValue(nil, "", fieldsMap)
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(fields, func(i, j int) bool { return fields[i].field < fields[j].field })
+	return fields, nil
```

Comment:
For arrays, will sorting do alphabetical sort as opposed to numerical "7" > "10"

Reply:
It might, good catch! I will fix this and add a test for it.